thermal.c

```diff
@@ -14,43 +14,333 @@
 
 cpumask_t thermal_banned_cpus;
 
+#define THERMAL_GENL_FAMILY_NAME	"thermal"
+#define THERMAL_GENL_EVENT_GROUP_NAME	"event"
+
+struct family_data {
+	const char *group;
+	int id;
+};
+
+static struct nl_sock *sock;
+static struct nl_cb *callback;
+
 static gboolean prepare_netlink(void)
 {
-	gboolean error = TRUE;
+	gboolean error = FALSE;
+	int rc;
+
+	sock = nl_socket_alloc();
+	if (!sock) {
+		log(TO_ALL, LOG_ERR, "thermal: socket allocation failed.\n");
+		goto err_out;
+	}
 
-	log(TO_ALL, LOG_ERR, "thermal: not yet implement to alloc memory for netlink.\n");
+	rc = genl_connect(sock);
+	if (rc) {
+		log(TO_ALL, LOG_ERR, "thermal: socket bind failed.\n");
+		goto err_out;
```

Comment:
memory leak, need to free the allocated socket

Reply:
There is a fall-though path: prepare_netlink()->deinit_thermal()->nl_socket_free(sock)